gooddata-sdk/gooddata_sdk/catalog/declarative/analytics_layer.py

```diff
@@ -0,0 +1,167 @@
+            ]
+
+        return cls(analytical_dashboards, dashboard_plugins, filter_contexts, metrics, visualization_objects)
+
+    def to_api(self) -> DeclarativeAnalyticsLayer:
+        kwargs: dict[str, Any] = {
+            "analytical_dashboards": [],
+            "dashboard_plugins": [],
+            "filter_contexts": [],
+            "metrics": [],
+            "visualization_objects": [],
+        }
+
+        if self.analytical_dashboards:
+            kwargs["analytical_dashboards"] = [v.to_api() for v in self.analytical_dashboards]
+        if self.dashboard_plugins:
+            kwargs["dashboard_plugins"] = [v.to_api() for v in self.dashboard_plugins]
+        if self.filter_contexts:
+            kwargs["filter_contexts"] = [v.to_api() for v in self.filter_contexts]
+        if self.metrics:
+            kwargs["metrics"] = [v.to_api() for v in self.metrics]
+        if self.visualization_objects:
+            kwargs["visualization_objects"] = [v.to_api() for v in self.visualization_objects]
+        return DeclarativeAnalyticsLayer(**kwargs)
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, CatalogDeclarativeAnalyticsLayer):
+            return False
+        return (
+            self.analytical_dashboards == other.analytical_dashboards
+            and self.dashboard_plugins == other.dashboard_plugins
+            and self.filter_contexts == other.filter_contexts
+            and self.metrics == other.metrics
+            and self.visualization_objects == other.visualization_objects
+        )
+
+
+class CatalogAnalyticsBase:
+    def __init__(self, id: str, title: str, content: dict[str, Any], description: str = None, tags: list[str] = None):
+        self.id = id
+        self.title = title
+        self.content = content
+        self.description = description
+        self.tags = tags
+
+    @classmethod
+    def from_api(cls: Type[T], entity: dict[str, Any]) -> T:
+        return cls(entity["id"], entity["title"], entity["content"], entity.get("description"), entity.get("tags"))
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, CatalogAnalyticsBase):
+            return False
+        return (
+            self.id == other.id
+            and self.title == other.title
+            and self.content == other.content
+            and self.description == other.description
+            and self.tags == other.tags
+        )
+
+    def __str__(self) -> str:
+        return self.__repr__()
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}(id={self.id}, title={self.title})"
+
+
+class CatalogDeclarativeAnalyticalDashboard(CatalogAnalyticsBase):
+    def to_api(self) -> DeclarativeAnalyticalDashboard:
+        kwargs: dict[str, Any] = dict()
+        if self.description:
+            kwargs["description"] = self.description
+        if self.tags:
+            kwargs["tags"] = self.tags
```

Comment:
Repetitive code. Create method to build `kwargs` in ancestor and use it on all places.

Reply:
Good note, thanks.